snc.sh

```diff
@@ -120,30 +120,30 @@ fi
 
 # Create the INSTALL_DIR for the installer and copy the install-config
 rm -fr ${INSTALL_DIR} && mkdir ${INSTALL_DIR} && cp install-config.yaml ${INSTALL_DIR}
-${YQ} write --inplace ${INSTALL_DIR}/install-config.yaml compute[0].architecture ${yq_ARCH}
-${YQ} write --inplace ${INSTALL_DIR}/install-config.yaml controlPlane.architecture ${yq_ARCH}
-${YQ} write --inplace ${INSTALL_DIR}/install-config.yaml baseDomain ${BASE_DOMAIN}
-${YQ} write --inplace ${INSTALL_DIR}/install-config.yaml metadata.name ${CRC_VM_NAME}
-${YQ} write --inplace ${INSTALL_DIR}/install-config.yaml compute[0].replicas 0
-${YQ} write --inplace ${INSTALL_DIR}/install-config.yaml pullSecret "@HIDDEN_PULL_SECRET@"
+yq_ARCH=${yq_ARCH} ${YQ} eval --inplace '.compute[0].architecture = strenv(yq_ARCH)' ${INSTALL_DIR}/install-config.yaml
+yq_ARCH=${yq_ARCH} ${YQ} eval --inplace '.controlPlane.architecture = strenv(yq_ARCH)' ${INSTALL_DIR}/install-config.yaml
+base_domain=${BASE_DOMAIN} ${YQ} eval --inplace '.baseDomain = strenv(base_domain)' ${INSTALL_DIR}/install-config.yaml
+crc_vm_name=${CRC_VM_NAME} ${YQ} eval --inplace '.metadata.name = strenv(crc_vm_name)' ${INSTALL_DIR}/install-config.yaml
+${YQ} eval --inplace '.compute[0].replicas = 0' ${INSTALL_DIR}/install-config.yaml
+${YQ} eval --inplace '.pullSecret = "@HIDDEN_PULL_SECRET@"' ${INSTALL_DIR}/install-config.yaml
 replace_pull_secret ${INSTALL_DIR}/install-config.yaml
-${YQ} write --inplace ${INSTALL_DIR}/install-config.yaml sshKey "$(cat id_ecdsa_crc.pub)"
+sshKey=$(cat id_ecdsa_crc.pub) ${YQ} eval '.sshKey = strenv(sshKey)' --inplace ${INSTALL_DIR}/install-config.yaml
 
 # Create the manifests using the INSTALL_DIR
 OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=$OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE ${OPENSHIFT_INSTALL} --dir ${INSTALL_DIR} create manifests || exit 1
 
 # Add CVO overrides before first start of the cluster. Objects declared in this file won't be created.
-${YQ} merge -i ${INSTALL_DIR}/manifests/cvo-overrides.yaml cvo-overrides.yaml
+${YQ} eval-all --inplace 'select(fileIndex == 0) * select(filename == "cvo-overrides.yaml")' ${INSTALL_DIR}/manifests/cvo-overrides.yaml cvo-overrides.yaml
```

Comment:
Some references to the new documentation in the commit log would be useful, https://mikefarah.gitbook.io/yq/v/v4.x/upgrading-from-v3#merging-documents for example.